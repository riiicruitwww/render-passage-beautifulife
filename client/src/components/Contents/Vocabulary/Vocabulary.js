import React from 'react';
import PropTypes from 'prop-types';
import './Vocabulary.scss';

const Vocabulary = ({ vocabulary }) => {
  return (
    <span className="Vocabulary">
      <span className="Vocabulary__word">{vocabulary.word}:</span>
      <span className="Vocabulary__meaning">{vocabulary.meaning}</span>
    </span>
  );
};

Vocabulary.propTypes = {
  vocabulary: PropTypes.shape({
    word: PropTypes.string.isRequired,
    meaning: PropTypes.string.isRequired
  }).isRequired
};

export default Vocabulary;
