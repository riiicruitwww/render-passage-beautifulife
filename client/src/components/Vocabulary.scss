.Vocabulary {
  display: inline-block;
  margin: 0 4rem 1rem 0;

  &__word {
    color: rgba(0, 0, 0, 0.9);
  }

  &__meaning {
    margin-left: 0.5rem;
  }
}
