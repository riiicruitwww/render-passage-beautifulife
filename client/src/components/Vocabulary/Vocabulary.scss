.Vocabulary {
  display: inline-block;
  margin: 0 64px 16px 0;

  &__word {
    color: rgba(0, 0, 0, 0.9);
  }

  &__meaning {
    margin-left: 8px;
  }
}
