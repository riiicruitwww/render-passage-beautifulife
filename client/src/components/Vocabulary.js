import React from 'react';
import './Vocabulary.scss';

const Vocabulary = ({ vocabulary }) => {
  return (
    <span className="Vocabulary">
      <span className="Vocabulary__word">{vocabulary.word}:</span>
      <span className="Vocabulary__meaning">{vocabulary.meaning}</span>
    </span>
  );
};

export default Vocabulary;
