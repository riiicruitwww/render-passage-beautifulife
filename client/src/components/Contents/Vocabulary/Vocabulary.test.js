import React from 'react';
import { shallow } from 'enzyme';
import Vocabulary from './Vocabulary';

describe('Vocabulary', () => {
  describe('about render', () => {
    const props = {
      vocabulary: {
        word: 'test',
        meaning: 'test'
      }
    };
    const wrapper = shallow(<Vocabulary {...props} />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
