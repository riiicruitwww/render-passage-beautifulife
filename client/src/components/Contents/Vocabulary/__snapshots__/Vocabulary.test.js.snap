// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vocabulary about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vocabulary
    vocabulary={
      Object {
        "meaning": "test",
        "word": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="Vocabulary__word"
        >
          test
          :
        </span>,
        <span
          className="Vocabulary__meaning"
        >
          test
        </span>,
      ],
      "className": "Vocabulary",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "test",
            ":",
          ],
          "className": "Vocabulary__word",
        },
        "ref": null,
        "rendered": Array [
          "test",
          ":",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "Vocabulary__meaning",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="Vocabulary__word"
          >
            test
            :
          </span>,
          <span
            className="Vocabulary__meaning"
          >
            test
          </span>,
        ],
        "className": "Vocabulary",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "test",
              ":",
            ],
            "className": "Vocabulary__word",
          },
          "ref": null,
          "rendered": Array [
            "test",
            ":",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "Vocabulary__meaning",
          },
          "ref": null,
          "rendered": "test",
          "type": "span",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
